### ISSUE TITLE
Can not reproduce the results

### ISSUE THREAD

**Issue by Boey-li** [ap229997/hands#3]
Hello, thanks for the great work!

I attempted to reproduce the results using the demo code and example data you provided; however, the outcomes were not as strong as those presented in your paper.

I replaced the hand detection component from ViTPoseModel with YOLOv8. Below is the visualization of the hand detection, which appears to be accurate.
![image](https://github.com/user-attachments/assets/1e88369f-9d18-4dfd-89d5-2da5601e81ca)
![image](https://github.com/user-attachments/assets/03080963-5d05-4513-9ef7-a08c69b3effe)

I used the detected bounding box and the is_right flag as inputs for the dataset, but the 3D hand reconstruction does not seem very accurate.
![image](https://github.com/user-attachments/assets/c6ac414c-f453-4597-82bf-605bb6549459)
![image](https://github.com/user-attachments/assets/a1e2ecd3-f292-4ffb-bee0-9dd830217b75)

Is this expected result? If not, what could be the potential causes for this issue?

**Comment by ap229997** [comment]
I have not tried with YOLO boxes, can you provide some more details:
- What is the format of the bounding boxes: `[x0, y0, x1, y1]` or `[x0, y0, w, h]` or something else?
- Is the rescale factor the same or different with the new bounding boxes?
- What is the input image resolution to the model?

I'll also check the demo code to see if there is any issue somewhere, arising due to different input formats.

In the meantime, I'd also suggest to try [HaMeR](https://geopavlakos.github.io/hamer/) or [WiLoR](https://rolpotamias.github.io/WiLoR/) to see if it works better for your use case.

**Comment by Boey-li** [comment]
I follow the hand detection procedures from **WiLoR** to get the hand bounding box with YOLO.
- The format of the bounding boxes should be `[x0, y0, x1, y1]`.
- I do not change any other parameters for the input, i.e., the rescale factor and image resolution should be the same as before. 
- I notice that the rescale factor for WiLoR is 2.5, would this matter?

For my case, I would like to get the absolute 3D coordinates for hand joint keypoints in the `camera coordinate system`. Both HaMeR and WiLoR do not take camera parameters / focal length as input and assume a focal length of 5000 in its predictions. Thus, there exists scale depth ambiguity between the estimated depth and actual depth. 

I have read your paper and see your method is conditioned on the camera intrinsic, but when I input the camera focal length, the estimated depth still far beyond the actual depth. 

Do you have any suggestions on solving the issues of scale depth ambiguity? Thanks a lot!

**Comment by ap229997** [comment]
I think this is partly related to issue #4 that you also raised. There is a lot of zero padding around the image due to the image not being square but the demo code expecting a square image. I'll update the demo code to account for this and modify the intrinsic accordingly before passing to the dataloader.

**Comment by ap229997** [comment]
I apologize for the long delay. It took me some time to get to this.

I updated the code to take intrinsics as input and modify it for arbitrary-sized images. This improves the position of the rendered hand to some extent but there is still quite a bit of error in the global pose of the wrist w.r.t camera. One reason I identified is that the model is sensitive to the input bounding box (this matters since the positional encoding depends on the bounding box around the hand). I tried varying bounding box input (predictions from different off-the-shelf models vs ground truth bounding box) and could see the variations in the global pose. The visualizations in the paper are done with ground truth bounding boxes. One way to fix this would be to use aggressive bounding box augmentation during training (we do use augmentation but that does not seem to be sufficient). As for exocentric images, our model won't work since it is not trained on any exocentric data.
